apollo-router/src/services/router/body.rs

```diff
@@ -1,133 +1,63 @@
-    }
-
-    pub fn wrap_stream<S, O, E>(stream: S) -> RouterBody
-    where
-        S: Stream<Item = Result<O, E>> + Send + 'static,
-        O: Into<Bytes> + 'static,
-        E: Into<Box<dyn std::error::Error + Send + Sync>> + 'static,
-    {
-        Self(super::Body::wrap_stream(stream))
-    }
+pub(crate) async fn get_body_bytes<B: HttpBody>(body: B) -> Result<Bytes, B::Error> {
+    Ok(body.collect().await?.to_bytes())
 }
 
-impl<T: Into<super::Body>> From<T> for RouterBody {
-    fn from(value: T) -> Self {
-        RouterBody(value.into())
-    }
-}
+// We create some utility functions to make Empty and Full bodies
+// and convert types
 
-impl Debug for RouterBody {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.0.fmt(f)
-    }
+/// Create an empty RouterBody
+pub(crate) fn empty() -> UnsyncBoxBody<Bytes, AxumError> {
+    Empty::<Bytes>::new()
+        .map_err(|never| match never {})
+        .boxed_unsync()
 }
 
-impl Stream for RouterBody {
-    type Item = <hyper::body::Body as Stream>::Item;
-
-    fn poll_next(
-        mut self: std::pin::Pin<&mut Self>,
-        cx: &mut std::task::Context<'_>,
-    ) -> std::task::Poll<Option<Self::Item>> {
-        let mut pinned = std::pin::pin!(&mut self.0);
-        pinned.as_mut().poll_next(cx)
-    }
+/// Create a Full RouterBody using the supplied chunk
+pub(crate) fn full<T: Into<Bytes>>(chunk: T) -> UnsyncBoxBody<Bytes, AxumError> {
+    Full::new(chunk.into())
+        .map_err(|never| match never {})
+        .boxed_unsync()
 }
 
-impl HttpBody for RouterBody {
-    type Data = <hyper::body::Body as HttpBody>::Data;
-
-    type Error = <hyper::body::Body as HttpBody>::Error;
-
-    fn poll_data(
-        mut self: std::pin::Pin<&mut Self>,
-        cx: &mut std::task::Context<'_>,
-    ) -> std::task::Poll<Option<Result<Self::Data, Self::Error>>> {
-        let mut pinned = std::pin::pin!(&mut self.0);
-        pinned.as_mut().poll_data(cx)
-    }
-
-    fn poll_trailers(
-        mut self: std::pin::Pin<&mut Self>,
-        cx: &mut std::task::Context<'_>,
-    ) -> std::task::Poll<Result<Option<http::HeaderMap>, Self::Error>> {
-        let mut pinned = std::pin::pin!(&mut self.0);
-        pinned.as_mut().poll_trailers(cx)
-    }
-
-    fn is_end_stream(&self) -> bool {
-        self.0.is_end_stream()
-    }
-
-    fn size_hint(&self) -> SizeHint {
-        HttpBody::size_hint(&self.0)
-    }
-}
-
-pub(crate) async fn get_body_bytes<B: HttpBody>(body: B) -> Result<Bytes, B::Error> {
-    hyper::body::to_bytes(body).await
-}
-
-// this is used to wrap a hyper::Client because its Service implementation will always return a hyper::Body,
-// it does not keep the body type used by the request
-#[derive(Clone)]
-pub(crate) struct RouterBodyConverter<C> {
-    pub(crate) inner: C,
+/// Create a streaming RouterBody using the supplied stream
+pub(crate) fn from_result_stream<S, E>(data_stream: S) -> RouterBody
```

Comment:
`from_data_stream` to match its opposite function, `into_data_stream`? 
```suggestion
pub(crate) fn from_data_stream<S, E>(data_stream: S) -> RouterBody
```

Reply:
I'm not going to make this change since it actually is a result stream that we are processing.